### PR TITLE
Add ForwardAgent, fixes concourse/git-resource#243

### PR DESCRIPTION
**Pull request by agrrh**
It's impossible to access git repos using SSH jump hosts.

See #243 for details.

Adding `ForwardAgent` option here, fixed issue for me.

### CODE REVIEW

**Comment by danilopopeye** [comment]
@agrrh maybe this can be a configuration flag?

**Comment by vito** [comment]
Just so I understand, is this so things work when your SSH URI is actually a proxy through to another SSH server? So there's no proxy configuration in the `git` resource itself?

**Comment by agrrh** [comment]
@vito Yes, you are right.

**Comment by vito** [comment]
@agrrh Sounds good. Would you mind adding a test for this? I think you could just add one similar to [this test](https://github.com/concourse/git-resource/blob/master/test/check.sh#L29-L44), and just `grep "ForwardAgent yes" $HOME/.ssh/config` in the test. Looks like you'd need to add a helper that sets `forward_host: true`, as well.

**Comment by agrrh** [comment]
@vito Added test and checked if it succeeds/fails when appropriate option set to true/false respectively.

File: test/check.sh [hunk 1]
```diff
@@ -42,6 +42,16 @@ it_fails_if_key_has_password() {
...
 }
 
+it_fails_if_ssh_agent_failed() {
+  local repo=$(init_repo)
+  local key=$TMPDIR/key-no-passphrase
+
+  ssh-keygen -f $key
+  check_uri_with_key_and_ssh_agent $repo $key
+
+  [ ! -z "$(grep "ForwardAgent yes" $HOME/.ssh/config)" ]
```
[the remark this answers is not in the document]
Comment (agrrh) [reply test/check.sh]: This version keeps "true/false" as result, keeping same style as in existsing tests:
```
running it_checks_given_paths...
...
  true

running it_checks_given_glob_paths...
...
  true
```
I don't mind to simplify final check as you propose, but in this case we would see grep result as test result:
```
running it_fails_if_ssh_agent_failed...
...
  ForwardAgent yes
```
Shouldn't we keep same true/false style here?

[the remark this answers is not in the document]
Comment (agrrh) [reply test/check.sh]: Oh, sorry, those true/false were `jq` output.

File: test/check.sh [hunk 2]
```diff
@@ -42,6 +42,16 @@ it_fails_if_key_has_password() {
   grep "Private keys with passphrases are not supported." $failed_output
 }
 
+it_fails_if_ssh_agent_failed() {
```
[the remark this answers is not in the document]
Comment (agrrh) [reply test/check.sh]: Nice catch, thank you. Renamed test as you proposed and added second "it_skips_forward_agent_configuration" test for case when we expect "disabled" state.

File: test/check.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (agrrh) [reply test/check.sh]: Fixed as you proposed.

**Comment by agrrh** [comment]
@vito Replaced with 2 separate tests:
- `it_configures_forward_agent`
- `it_skips_forward_agent_configuration`

**Comment by vito** [comment]
@agrrh Thanks, looks good now! 👍

**Pull request merged**